src/GafferScene/ShaderPlug.cpp

```diff
@@ -48,31 +48,23 @@ using namespace Gaffer;
 using namespace GafferScene;
 
 //////////////////////////////////////////////////////////////////////////
-// Internal utilities
+// ShaderPlug
 //////////////////////////////////////////////////////////////////////////
 
-namespace
-{
+IE_CORE_DEFINERUNTIMETYPED( ShaderPlug );
 
-const GafferScene::Shader *shader( const ShaderPlug *plug )
+
+const Gaffer::Plug *ShaderPlug::getShaderOutPlug() const
```

Comment:
Made this a member function as it seemed a little strange that it's only parameter was this. Perhaps this was done so the public header didn't have to change?